### DISCUSSION TITLE
AP2 Build on ArchLinux

### DISCUSSION THREAD

**Discussion by ckdo** [mikebrady/shairport-sync#1234] [General]
Hello,

I'm trying to compile on archlinux.
I had to alter configure.ac so that libplist is found ( "-2.0" added )

PKG_CHECK_MODULES([libplist], [libplist-2.0 >= 2.0.0]

Then I have this error:

@mikebrady any idea ? 

![image](https://user-images.githubusercontent.com/15435601/125322789-6c563a80-e33e-11eb-9520-5b11fdd8e404.png)

**Comment by mikebrady** [comment]
Hi @ckdo. There is some issue with `libplist` in Alpine -- it doesn't seem to work. Have a look at the Docker file, which is pinned to Alpine 3:12:

https://github.com/mikebrady/shairport-sync/blob/development/docker/Dockerfile

**Reply by charlesomer** [reply]
> Breaking
> * Rename library and all related files by adding an API version resulting in libplist-2.0 and libplist++-2.0

This is in the release notes for v2.2.0 (https://github.com/libimobiledevice/libplist/releases). v2.2.0 is being included in Debian Bullseye (and is what's used in Alpine v3.13 and above) so once that gets released I think that will break Shairport Sync as it currently stands - but that's probably a while off yet fortunately :)

**Reply by ckdo** [reply]
Hello @charlesomer ,

Seems to be due to that 2.2.0 release, this one is packaged with last ArchLinux Arm. I'll try with a locally compiled lower version.

**Reply by ckdo** [reply]
I've tried with libplist 2.1.0 compiled but I got the same error.

Where this plists_get_info_response_xml & plists_get_info_response_xml_len are supposed to be declared ?

**Comment by mikebrady** [comment]
I'm actually trying to get your code running too!

**Reply by ckdo** [reply]
@jdtsmith I've opened "discussions" on github if have any trouble with running ckdo/airplay2-receiver

**Comment by jdtsmith** [comment]
From the other direction, I had a `libplist` (v1.12) issue with Rasbian Stretch and had to upgrade to Buster (v2.0.1).

**Reply by alexfry** [reply]
I think I was getting caught by this same issue. Updating to Buster now.

**Reply by ckdo** [reply]
I have libplist 2.2.0 so I think this is due to "breaking" notes from this specific version
